internal/testplanet/satellite.go

```diff
@@ -4,38 +4,195 @@
 package testplanet
 
 import (
+	"context"
+	"net"
 	"os"
 	"path/filepath"
 	"strconv"
 	"time"
 
 	"github.com/zeebo/errs"
+	"go.uber.org/zap"
+	"golang.org/x/sync/errgroup"
 
+	"storj.io/storj/internal/errs2"
 	"storj.io/storj/internal/memory"
+	"storj.io/storj/internal/version"
+	"storj.io/storj/pkg/identity"
 	"storj.io/storj/pkg/peertls/extensions"
 	"storj.io/storj/pkg/peertls/tlsopts"
 	"storj.io/storj/pkg/revocation"
+	"storj.io/storj/pkg/rpc"
 	"storj.io/storj/pkg/server"
+	"storj.io/storj/pkg/storj"
 	"storj.io/storj/satellite"
+	"storj.io/storj/satellite/accounting"
+	"storj.io/storj/satellite/accounting/live"
 	"storj.io/storj/satellite/accounting/rollup"
 	"storj.io/storj/satellite/accounting/tally"
 	"storj.io/storj/satellite/audit"
 	"storj.io/storj/satellite/console"
 	"storj.io/storj/satellite/console/consoleweb"
+	"storj.io/storj/satellite/contact"
 	"storj.io/storj/satellite/dbcleanup"
 	"storj.io/storj/satellite/discovery"
 	"storj.io/storj/satellite/gc"
 	"storj.io/storj/satellite/gracefulexit"
+	"storj.io/storj/satellite/inspector"
 	"storj.io/storj/satellite/mailservice"
 	"storj.io/storj/satellite/marketingweb"
 	"storj.io/storj/satellite/metainfo"
+	"storj.io/storj/satellite/nodestats"
 	"storj.io/storj/satellite/orders"
 	"storj.io/storj/satellite/overlay"
 	"storj.io/storj/satellite/repair/checker"
+	"storj.io/storj/satellite/repair/irreparable"
 	"storj.io/storj/satellite/repair/repairer"
 	"storj.io/storj/satellite/satellitedb"
+	"storj.io/storj/satellite/vouchers"
 )
 
+// SatelliteSystem contains all the processes needed to run a full Satellite setup
+type SatelliteSystem struct {
+	Peer *satellite.Peer
+	API  *satellite.API
+
+	Log      *zap.Logger
+	Identity *identity.FullIdentity
+	DB       satellite.DB
+
+	Dialer rpc.Dialer
+
+	Server *server.Server
+
+	Version *version.Service
+
+	Contact struct {
```

Comment:
Do we need all of these services added to the `SatelliteSystem`? Aren't they already in the `Peer/API`?

Reply:
Adding all these to the SatelliteSystem makes it so that I don't need to change every test.  If I didn't do this, then I would have to manually go through and update all test to either use the `SatelliteSystem.API` or `SatelliteSystem.Peer`.  Long term we will want to make that change, but I'm planning on doing it in another PR since it will be a lot of work.